Add warning alert for too high distributor GC CPU utilization (#10641)
grafana/mimir: docs/sources/mimir/manage/mimir-runbooks/_index.md

1. Ensure shuffle-sharding is enabled in the Mimir cluster
1. Assuming shuffle-sharding is enabled, scaling up ingesters will lower the number of tenants per ingester. However, the effect of this change will be visible only after `-blocks-storage.tsdb.close-idle-tsdb-timeout` period so you may have to temporarily increase the limit

### MimirDistributorGcUsesTooMuchCpu

This alert fires when distributors spend too much CPU time on garbage collection.

> This should be a symptom of GOMEMLIMIT being set too low, so GC gets triggered too frequently.

Reviewer: ```suggestion
This is a symptom of setting GOMEMLIMIT too low, so GC is triggered too frequently.
```
